File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/al_to_am.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

// Ham chuyen danh sach ke sang ma tran ke
vector<vector<int>> alToAm(const vector<vector<int>>& adjList) {
    int n = adjList.size();
    vector<vector<int>> matrix(n, vector<int>(n, 0)); // Khoi tao ma tran 0

    for (int u = 0; u < n; ++u) {
        for (int v : adjList[u]) {
            matrix[u][v] = 1; // Danh dau canh tu u den v
        }
    }

    return matrix;
}

int main() {
    int n;
    cout << "Nhap so dinh: ";
    cin >> n;
    cin.ignore(); // Loai bo ky tu xuong dong

    vector<vector<int>> adjList(n);

    cout << "Nhap danh sach ke (AL):\n";
    for (int i = 0; i < n; ++i) {
        cout << i << ": ";
        string line;
        getline(cin, line);
        stringstream ss(line);
        int v;
        while (ss >> v)
            adjList[i].push_back(v);
    }

    auto matrix = alToAm(adjList);

    cout << "Ma tran ke (AM):\n";
    for (const auto& row : matrix) {
        for (int val : row)
            cout << val << " ";
        cout << "\n";
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/al_to_am.py
def al_to_am(adj_list):
    n = len(adj_list)
    matrix = [[0]*n for _ in range(n)]  # Khoi tao ma tran toan 0

    for u in range(n):
        for v in adj_list[u]:
            matrix[u][v] = 1  # Danh dau canh u -> v

    return matrix

# Nhap tu ban phim
n = int(input("Nhap so dinh: "))
adj_list = []
print("Nhap danh sach ke (AL):")
for i in range(n):
    row = list(map(int, input(f"{i}: ").split()))
    adj_list.append(row)

am = al_to_am(adj_list)

print("Ma tran ke (AM):")
for row in am:
    print(' '.join(map(str, row)))